src/main/java/com/itrex/java/lab/service/impl/FillingDatabaseTestServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+                        .build();
+                userList.add(user);
+                userNumber++;
+            }
+            userRepository.saveAll(userList);
+            return true;
+        }
+
+        if (Role.CONTRACTOR.name().equals(role)) {
+            List<User> userList = new ArrayList<>();
+            for (int i = 0; i < quantity; i++) {
+                User user = User.builder()
+                        .name(BASE_CONTRACTOR_NAME + userNumber)
+                        .password("$2a$12$ma0C6gCWmZL/d3FSt78mF.FCcLdfBqc4XXyRrDBv54EbWEPcj3OIC")
+                        .role(Role.CONTRACTOR)
+                        .status(Status.ACTIVE)
+                        .email(BASE_CONTRACTOR_NAME + userNumber + "@gmail.com")
+                        .build();
+                userList.add(user);
+                userNumber++;
+            }
+            userRepository.saveAll(userList);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean fillContractTableForEachCustomer(int minContractAmount, int maxContractAmount) {
+        List<Contract> contracts = new ArrayList<>();
+        Iterable<User> contractOwners = userRepository.findByRole(Role.CUSTOMER);
+
+        for (User user : contractOwners) {
+            contracts.addAll(createContracts(user, minContractAmount, maxContractAmount));
+        }
+        contractRepository.saveAll(contracts);
+        return true;
+    }
+
+    private List<Contract> createContracts(User user, int minContractAmount, int maxContractAmount) {
+        List<Contract> contracts = new ArrayList<>();
+        int contractAmount = RANDOM_GENERATOR.nextInt(maxContractAmount - minContractAmount) + minContractAmount;
+        for (int i = 0; i < contractAmount; i++) {
+            Contract contract = Contract.builder()
+                    .owner(user)
+                    .description("Contract" + i)
+                    .startDate(LocalDate.now().plusDays(RANDOM_GENERATOR.nextInt(90)))
+                    .endDate(LocalDate.now().plusDays(100).plusDays(RANDOM_GENERATOR.nextInt(90)))
+                    .startPrice(RANDOM_GENERATOR.nextInt(10000) + 20000)
+                    .build();
+            contracts.add(contract);
+        }
+        return contracts;
+    }
+
+    @Override
+    public boolean fillOffersForEachContract(int minOfferAmount, int maxOfferAmount) {
+        Iterable<Contract> contracts = contractRepository.findAll();
+        Iterable<User> allContractors = userRepository.findByRole(Role.CONTRACTOR);
+        List<User> contractors = new ArrayList<>();
+        allContractors.forEach(contractors::add);
+
+        for (Contract contract : contracts) {
+            List<Offer> offers = new ArrayList<>();
+            offers.addAll(createOffers(contract, minOfferAmount, maxOfferAmount, contractors));
+            offerRepository.saveAll(offers);
+        }
+        return true;
```

Comment:
offerRepository.saveAll(offers)

not critical, but better to read.
1. first step  - prepare data
2. second step - save data

Reply:
Done.